mtls-certificates: CertificateRoleMappingTest fails in native

### Describe the bug

```
[ERROR] Failures: 
[ERROR]   CertificateRoleMappingIT>CertificateRoleMappingTest.testNoClientCertificate:49 Insecure requests must fail at the transport level ==> Unexpected exception type thrown, expected: <java.net.ConnectException> but was: <javax.net.ssl.SSLHandshakeException>
```


### Expected behavior

Both JVM mode and Native mode pass the test.

### Actual behavior

Native mode fails.

### How to Reproduce?

```
$ ./mvnw clean install -f integration-tests -pl mtls-certificates -Pnative -Dstart-containers -DskipTests=false -DskipITs=false -Dnative.surefire.skip=false

```

### Output of `uname -a` or `ver`

Linux mandrel2 5.14.0-407.el9.aarch64 #1 SMP PREEMPT_DYNAMIC Thu Jan 11 00:18:37 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux

### Output of `java -version`

OpenJDK Runtime Environment Mandrel-23.1.3.1-Final (build 21.0.3+9-LTS)

### Quarkus version or git rev

3.10.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.3 (21122926829f1ead511c958d89bd2f672198ae9f)

### Additional information

_No response_

--- a/integration-tests/mtls-certificates/src/main/resources/application.properties
+++ b/integration-tests/mtls-certificates/src/main/resources/application.properties
@@ -4,5 +4,4 @@ quarkus.http.ssl.certificate.trust-store-file=server-truststore.p12
 quarkus.http.ssl.certificate.trust-store-password=password
 quarkus.http.ssl.client-auth=REQUIRED
 quarkus.http.auth.certificate-role-properties=cn-role-mappings.txt
-quarkus.native.additional-build-args=-H:IncludeResources=.*\\.p12,-H:IncludeResources=.*\\.txt
-
+quarkus.native.resources.includes=*.p12,*.txt

--- a/integration-tests/mtls-certificates/src/test/java/io/quarkus/it/vertx/AbstractCertificateRoleMappingTest.java
+++ b/integration-tests/mtls-certificates/src/test/java/io/quarkus/it/vertx/AbstractCertificateRoleMappingTest.java
@@ -4,8 +4,9 @@ import static io.restassured.RestAssured.given;
 import static org.hamcrest.Matchers.equalTo;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
-import java.net.ConnectException;
 import java.net.URL;
+
+import javax.net.ssl.SSLHandshakeException;
 
 import org.junit.jupiter.api.Test;
 
@@ -44,19 +45,25 @@ public abstract class AbstractCertificateRoleMappingTest {
 
     @Test
     public void testNoClientCertificate() {
-        assertThrows(ConnectException.class,
-                () -> given().get("/protected/authenticated"),
+        // javax.net.ssl.SSLHandshakeException
+        // Indicates that the client and server could not negotiate the desired level of security.
+        // The connection is no longer usable.
+        final RequestSpecification rs = new RequestSpecBuilder()
+                .setBaseUri(String.format("%s://%s", url.getProtocol(), url.getHost()))
+                .setPort(url.getPort()).build();
+        assertThrows(SSLHandshakeException.class,
+                () -> given().spec(rs).get("/protected/authenticated"),
                 "Insecure requests must fail at the transport level");
-        assertThrows(ConnectException.class,
-                () -> given().get("/protected/authorized-user"),
+        assertThrows(SSLHandshakeException.class,
+                () -> given().spec(rs).get("/protected/authorized-user"),
                 "Insecure requests must fail at the transport level");
-        assertThrows(ConnectException.class,
-                () -> given().get("/protected/authorized-admin"),
+        assertThrows(SSLHandshakeException.class,
+                () -> given().spec(rs).get("/protected/authorized-admin"),
                 "Insecure requests must fail at the transport level");
     }
 
     protected RequestSpecification getMtlsRequestSpec(String clientKeyStore) {
-        var builder = new RequestSpecBuilder()
+        final RequestSpecBuilder builder = new RequestSpecBuilder()
                 .setBaseUri(String.format("%s://%s", url.getProtocol(), url.getHost()))
                 .setPort(url.getPort());
         withKeyStore(builder, clientKeyStore);

